libraries/AC_PrecLand/AC_PrecLand_Backend.h

```diff
@@ -28,16 +32,22 @@ class AC_PrecLand_Backend
 
     // provides a unit vector towards the target in body frame
     //  returns same as have_los_meas()
-    virtual bool get_los_body(Vector3f& dir_body) = 0;
+    virtual bool get_los_body(Vector3f& dir_body) {
+        if (have_los_meas()) {
+            dir_body = _los_meas_body;
+            return true;
+        }
+        return false;
+    };
 
     // returns system time in milliseconds of last los measurement
-    virtual uint32_t los_meas_time_ms() = 0;
+    virtual uint32_t los_meas_time_ms() { return _los_meas_time_ms; };
```

Comment:
Do these still need to be virtual?

Reply:
No, good catch. Should I make them final?

Reply:
Suite yourself on that one.  I've never actually found it to provide us an advantage.